### PR TITLE
Add ReversedGradient layer

### PR DESCRIPTION
**Pull request by PeterChe1990**
An easy way to have a layer which reverses the gradients of all weights before that layer. This can be useful in training models for e.g., domain-adversarial adaptation. #3119 and #4031 attempted to handle the same issue and this one simply uses `stop_gradient` borrowed from https://github.com/tensorflow/tensorflow/issues/4342.

### CODE REVIEW

**Comment by fchollet** [comment]
Thanks for the PR.

This is a valuable layer but its usefulness seems to be limited to a small niche of use cases. The cost of permanently adding it to the core Keras API (with all guarantees it entails) would outweigh the benefits. Instead, I recommend that you try to add this layer to `keras-contrib`, which exists precisely for this kind of functionality (with lesser maintenance guarantees).
